### PR TITLE
feat: added service account api

### PR DESCRIPTION
**Pull request by jackdelahunt**
## Description
This pr adds endpoints for the service account SDK. You can create, get and delete service accounts with these changes.

### Terraform config (main.tf)
```
terraform {
  required_providers {
    rhoas = {
      source  = "registry.terraform.io/redhat-developer/rhoas"
      version = "0.1.0"
    }
  }
}

provider "rhoas" {
    offline_token = "..."
}

resource "rhoas_service_account" "srvcaccnt" {
  name = "service_account"
}
```
### Commands
For mock server:
- `yarn`
- `yarn build`
- `yarn start`
For terraform provider:
- `make clean`
- `make install`
- `terraform init`
- `LOCAL_DEV=http://localhost:8000 terraform apply`

### Steps
- Run the above commands
- Then delete service account resource in `main.tf`
- Run `LOCAL_DEV=http://localhost:8000 terraform apply` again
- Service account should be deleted

### Expected Results
The service account should be created and then deleted

### CODE REVIEW

**Comment by jackdelahunt** [comment]
The verify steps are in terraform as they were the easiest for me. But just calling the API should work aswell

File: packages/api-mock/src/handlers/service-account-manager.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const nanoid = require("nanoid").nanoid;
+const path = require("path");
+const { getFullHostname } = require("../utls/host");
+
+let serviceAccountMap = new Map();
+
+const commonServiceAccountFields = {
+    id: nanoid(),
+    clientId: (Number.MAX_SAFE_INTEGER - new Date().getTime()).toString(),
```
Comment (wtrocki) [inline packages/api-mock/src/handlers/service-account-manager.js:9]: We use uuid lib in the mock. Feel free to use it

File: packages/api-mock/src/index.js [hunk 2]
```diff
@@ -86,21 +97,29 @@ srsControlApi.validateDefinition = () => {};
...
 srsDataApi.init();
 cosAPI.init();
+serviceAccountAPI.init();
 
 api.use("/api/accounts_mgmt/v1", amsRouter);
 
 api.use((req, res) => {
   if (req.url.startsWith("/api/kafkas_mgmt/v1")) {
     console.info("Calling kafkas manager");
+    console.log(req.headers)
```
Comment (wtrocki) [inline packages/api-mock/src/index.js:114]: we need to remove this

File: packages/api-mock/src/handlers/service-account-manager.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    createdBy: "service-account-created-by",
+    createdAt: new Date().getTime(),
+}
+
+function createServiceAccountHandlers(preSeed) {
+    
+    return {
+      createServiceAccount: async (c, req, res) => {
+
+        let serviceAccount = {
```
Comment (dimakis) [inline packages/api-mock/src/handlers/service-account-manager.js:22]: `serviceAccount` is a map here right? 

To keep this as close to the API as possible, the return object should be of type `ServiceAccountData` [as per here](https://github.com/redhat-developer/app-services-sdk-go/blob/main/serviceaccountmgmt/apiv1/client/docs/ServiceAccountsApi.md#return-type)

this passes and works with TF but fails with Ansible, as ansible expects the `ServiceAccountData` object type.

File: packages/api-mock/src/handlers/service-account-manager.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          name: req.body.name,
+          description: req.body.description,
+          ...commonServiceAccountFields
+        }
+
+        res.status(200).json(serviceAccount);
```
Comment (dimakis) [inline packages/api-mock/src/handlers/service-account-manager.js:28]: Does this only return a 200? there is no option to return a 400 or something as per the `getServiceAccount` below?

**Review by dimakis — COMMENTED** [review]
Just left a couple of comments

File: packages/api-mock/src/handlers/service-account-manager.js (hunk 3 above)
Comment (jackdelahunt) [reply packages/api-mock/src/handlers/service-account-manager.js:22]: the serviceAccount here is not a map it is a javascript object and its fields are added separately in the response body not as a map or anything likge that.

The docs give each field and its types which this response complies.

id, clientId and secreat are strings as nanoid returns a string, and the getTime func in the data object is a number everything else is of course a string so this maps correctly to the spec.

File: packages/api-mock/src/handlers/service-account-manager.js (hunk 4 above)
Comment (jackdelahunt) [reply packages/api-mock/src/handlers/service-account-manager.js:28]: only need to return other then 200 if something fails so i will check for the fields required in the body and if they are missing return the correct error code else 200 is fine.

**Comment by wtrocki** [comment]
Very good job!

**Pull request merged**